src/main/java/task/TaskList.java

```diff
@@ -0,0 +1,120 @@
+        this.todoList = todoList;
+    }
+
+    public void add(Task t) {
+        this.todoList.add(t);
+    }
+
+    public void printList() {
+        int length = this.todoList.size();
+        if (length == 0) {
+            System.out.println("There is no task in your list.");
+        } else {
+            System.out.println("Here are the tasks in your list:");
+
+            for (int i = 0; i < length; i++) {
+                String pos = String.valueOf(i + 1);
+                System.out.println(pos + "." + this.todoList.get(i));
+            }
+        }
+    }
+
+    private void showResult(TaskList result, String keyword) {
+        int length = result.todoList.size();
+        if (length == 0) {
+            System.out.println("There is no task in your list that contains the keyword '" + keyword + "'");
+        } else {
+            for (int i = 0; i < length; i++) {
+                String pos = String.valueOf(i + 1);
+                System.out.println(pos + "." + result.todoList.get(i));
+            }
+        }
+    }
+
+    /**
+     * List the whole TodoList after adding any tasks.
+     * 
+     * @param t
+     * @param storage
+     */
+
+    public void listOverviewAfterAdding(Task t, Storage storage) {
+        System.out.println("Got it. I've added this task:");
+        System.out.println(t);
+        int length = this.todoList.size();
+        if (length == 1) {
+            System.out.println("Now you have " + length + " task in the list.");
+        } else {
+            System.out.println("Now you have " + length + " tasks in the list.");
+        }
+        storage.autoUpdate(this.todoList);
+    }
+
+    /**
+     * This function handles the change of marking of tasks.
+     * It can mark task and unmark task.
+     * 
+     * @param userInput
+     * @param storage
+     */
+
+    public void changeMarkingOfTask(String userInput, Storage storage) {
+        String[] words = userInput.split("\\s+");
+        int number = Integer.parseInt(words[1]);
+        Task t = this.todoList.get(number - 1);
+        if (words[0].equals("mark")) {
+            t.markAsDone();
+        } else {
+            t.unmark();
+        }
+        storage.autoUpdate(this.todoList);
+    }
+
+    /**
+     * This function handle the deletion of task in the TodoList.
+     * 
+     * @param userInput
+     * @param storage
+     */
+
+    public void deleteTask(String userInput, Storage storage) {
+        String[] words = userInput.split("\\s+");
+        int number = Integer.parseInt(words[1]);
+        Task t = this.todoList.remove(number - 1);
+        System.out.println("Noted. I've removed this task:");
+        System.out.println(t);
+        int length = this.todoList.size();
+        if (length == 1) {
+            System.out.println("Now you have " + length + " task in the list.");
+        } else {
+            System.out.println("Now you have " + length + " tasks in the list.");
+        }
```

Comment:
Perhaps you could move this to Ui.java? So that all in/out can be handled there :)

Reply:
Ah I see! Thank you for pointing out!